contrib/dropwizard5/src/main/java/io/opencensus/contrib/dropwizard5/DropWizardMetrics.java

```diff
@@ -0,0 +1,332 @@
+import io.opencensus.metrics.export.TimeSeries;
+import io.opencensus.metrics.export.Value;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import javax.annotation.Nullable;
+
+/**
+ * Collects DropWizard metrics from a list {@link io.dropwizard.metrics5.MetricRegistry}s.
+ *
+ * <p>A {@link io.opencensus.metrics.export.MetricProducer} that wraps a DropWizardMetrics.
+ *
+ * @since 0.19
+ */
+public class DropWizardMetrics extends MetricProducer {
+
+  @DefaultVisibilityForTesting static final String DEFAULT_UNIT = "1";
+  @DefaultVisibilityForTesting static final String NS_UNIT = "ns";
+  private final List<io.dropwizard.metrics5.MetricRegistry> metricRegistryList;
+  private final MetricFilter metricFilter;
+  private final Clock clock;
+  private final Timestamp cumulativeStartTimestamp;
+
+  /**
+   * Hook the Dropwizard registry into the OpenCensus registry.
+   *
+   * @param metricRegistryList a list of {@link io.dropwizard.metrics5.MetricRegistry}s.
+   * @since 0.19
+   */
+  public DropWizardMetrics(List<io.dropwizard.metrics5.MetricRegistry> metricRegistryList) {
+    this(metricRegistryList, MetricFilter.ALL);
+  }
+
+  /**
+   * Hook the Dropwizard registry into the OpenCensus registry.
+   *
+   * @param metricRegistryList a list of {@link io.dropwizard.metrics5.MetricRegistry}s.
+   * @param metricFilter a filter to choose which metric to export
+   * @since 0.19
+   */
+  public DropWizardMetrics(
+      List<io.dropwizard.metrics5.MetricRegistry> metricRegistryList, MetricFilter metricFilter) {
+    Utils.checkListElementNotNull(
+        Utils.checkNotNull(metricRegistryList, "metricRegistryList"), "metricRegistry");
+    this.metricRegistryList = metricRegistryList;
+    this.metricFilter = Utils.checkNotNull(metricFilter, "metricFilter");
+    clock = MillisClock.getInstance();
+
+    // TODO(mayurkale): consider to add cache map<string, CacheEntry> where CacheEntry is
+    // {MetricDescriptor, startTime}
+    cumulativeStartTimestamp = clock.now();
+  }
+
+  /**
+   * Returns a {@code Metric} collected from {@link Gauge}.
+   *
+   * @param dropwizardMetric the metric name.
+   * @param gauge the gauge object to collect.
+   * @return a {@code Metric}.
+   */
+  @SuppressWarnings("rawtypes")
+  @Nullable
+  private Metric collectGauge(MetricName dropwizardMetric, Gauge gauge) {
+    // TODO cache dropwizard MetricName -> OC MetricDescriptor, Labels conversion
+    String metricName = DropWizardUtils.generateFullMetricName(dropwizardMetric.getKey(), "gauge");
+    String metricDescription =
+        DropWizardUtils.generateFullMetricDescription(dropwizardMetric.getKey(), gauge);
+    List<LabelValue> labelValues = new ArrayList<>();
+    List<LabelKey> labelKeys = new ArrayList<>();
+    for (Map.Entry<String, String> tag : dropwizardMetric.getTags().entrySet()) {
```

Comment:
I think it will be nice to have separate function to handle this conversion. Same everywhere.